src/crepe/system/InputSystem.cpp

```diff
@@ -155,48 +155,48 @@ void InputSystem::handle_non_mouse_event(const EventData & event) {
 
 void InputSystem::handle_move(const EventData & event_data, const vec2 & mouse_pos) {
 	ComponentManager & mgr = this->mediator.component_manager;
-
+	EventManager & event_mgr = this->mediator.event_manager;
 	RefVector<Button> buttons = mgr.get_components_by_type<Button>();
 
 	for (Button & button : buttons) {
 		if (!button.active) continue;
 		RefVector<Transform> transform_vec
 			= mgr.get_components_by_id<Transform>(button.game_object_id);
-		Transform & transform(transform_vec.front().get());
-
+		Transform & transform = transform_vec.front().get();
+		RefVector<Metadata> metadata_vec
+			= mgr.get_components_by_id<Metadata>(button.game_object_id);
+		Metadata & metadata = metadata_vec.front().get();
```

Comment:
Deze regels mogen direct vervangen worden met
```cpp
Metadata & metadata = mgr.get_components_by_id<Metadata>(button.game_object_id).front();
Transform & transform = mgr.get_components_by_id<Transform>(button.game_object_id).front();
```
Omdat GameObject al garandeert dat `.front()` bestaat voor `Metadata` en `Transform`, en door ze direct op te slaan in een reference hoef je geen `.get()` aan te roepen.

Dit wordt vaker gedaan in dit systeem dus verander het overal